google-cloud-spanner/src/test/java/com/google/cloud/spanner/MultiplexedSessionDatabaseClientMockServerTest.java

```diff
@@ -467,6 +476,106 @@ public void testWriteAtLeastOnceWithExcludeTxnFromChangeStreams() {
     assertEquals(1L, client.multiplexedSessionDatabaseClient.getNumSessionsReleased().get());
   }
 
+  // TODO(sriharshach): Uncomment test once Lock order preservation proto is published
+  /*
+    @Test
+    public void testAbortedReadWriteTxnUsesPreviousTxnIdOnRetryWithInlineBegin() throws InterruptedException {
+      DatabaseClientImpl client =
+          (DatabaseClientImpl) spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+      // Force the Commit RPC to return Aborted the first time it is called. The exception is cleared
+      // after the first call, so the retry should succeed.
+      mockSpanner.setCommitExecutionTime(
+          SimulatedExecutionTime.ofException(
+              mockSpanner.createAbortedException(ByteString.copyFromUtf8("test"))));
+      Thread.sleep(10000);
+      TransactionRunner runner = client.readWriteTransaction();
+      runner.run(
+          transaction -> {
+            try (ResultSet resultSet =
+                transaction.executeQuery(STATEMENT)) {
+              while (resultSet.next()) {}
+            }
+            return null;
+          });
+
+      List<ExecuteSqlRequest> executeSqlRequests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+      assertEquals(2, executeSqlRequests.size());
+
+      // Verify the requests are executed using multiplexed sessions
+      for (ExecuteSqlRequest request : executeSqlRequests) {
+        assertTrue(mockSpanner.getSession(request.getSession()).getMultiplexed());
+      }
+
+      // Verify that the first request uses inline begin, and the previous transaction ID is set to ByteString.EMPTY
+      assertTrue(executeSqlRequests.get(0).hasTransaction());
+      assertTrue(executeSqlRequests.get(0).getTransaction().hasBegin());
+      assertTrue(executeSqlRequests.get(0).getTransaction().getBegin().hasReadWrite());
+      assertNotNull(executeSqlRequests.get(0).getTransaction().getBegin().getReadWrite()
+          .getMultiplexedSessionPreviousTransactionId());
+      assertThat(executeSqlRequests.get(0).getTransaction().getBegin().getReadWrite().getMultiplexedSessionPreviousTransactionId()).isEqualTo(ByteString.EMPTY);
+
+      // Verify that the second request uses inline begin, and the previous transaction ID is set appropriately
+      assertTrue(executeSqlRequests.get(1).hasTransaction());
+      assertTrue(executeSqlRequests.get(1).getTransaction().hasBegin());
+      assertTrue(executeSqlRequests.get(1).getTransaction().getBegin().hasReadWrite());
+      assertNotNull(executeSqlRequests.get(1).getTransaction().getBegin().getReadWrite()
+          .getMultiplexedSessionPreviousTransactionId());
+      assertThat(executeSqlRequests.get(1).getTransaction().getBegin().getReadWrite().getMultiplexedSessionPreviousTransactionId()).isNotEqualTo(ByteString.EMPTY);
```

Comment:
Can we modify this test so that it not only checks that it is not empty, but that it has exactly the prevTransactionId that we expect (so equal to the transaction ID of the previous attempt).

Reply:
Done